discovery/openstack/instance.go

```diff
@@ -46,20 +46,21 @@ const (
 // InstanceDiscovery discovers OpenStack instances.
 type InstanceDiscovery struct {
 	authOpts *gophercloud.AuthOptions
+        listOpts *servers.ListOpts
 	region   string
 	interval time.Duration
 	logger   log.Logger
 	port     int
 }
 
 // NewInstanceDiscovery returns a new instance discovery.
-func NewInstanceDiscovery(opts *gophercloud.AuthOptions,
-	interval time.Duration, port int, region string, l log.Logger) *InstanceDiscovery {
+func NewInstanceDiscovery(opts *gophercloud.AuthOptions, 
+	listOpts *servers.ListOpts, interval time.Duration, port int, region string, l log.Logger) *InstanceDiscovery {
```

Comment:
I would rather pass the boolean value to `NewInstanceDiscovery()` and build `servers.ListOpts` here.